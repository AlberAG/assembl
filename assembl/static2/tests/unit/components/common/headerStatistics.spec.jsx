import React from 'react';
import renderer from 'react-test-renderer';

import HeaderStatistics, {
  statContributions,
  statMessages,
  statParticipants
} from '../../../../js/app/components/common/headerStatistics';
import '../../../helpers/setupTranslations';

describe('HeaderStatistics component', () => {
  it('should match HeaderStatistics snapshot', () => {
    const statElements = [statMessages(3), statContributions(5), statParticipants(2)];
    const component = renderer.create(<HeaderStatistics statElements={statElements} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});