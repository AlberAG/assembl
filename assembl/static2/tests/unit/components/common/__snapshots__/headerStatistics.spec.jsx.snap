// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderStatistics component should match HeaderStatistics snapshot 1`] = `
<div
  className="statistic"
>
  <div
    className="intermediary-container"
  >
    <div
      className="stat-container"
    >
      <div
        className="stat-box"
      >
        <div
          className="stat-icon assembl-icon-message white"
        />
        <div
          className="stat"
        >
          <div
            className="stat-nb"
          >
            <span
              className={undefined}
              style={undefined}
            >
              10,000
            </span>
          </div>
          <div
            className="stat-nb stat-label"
          >
            <span
              className={undefined}
              style={undefined}
            >
              messages
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="stat-container"
    >
      <div
        className="stat-box"
      >
        <div
          className="stat-icon assembl-icon-sentiment-neutral white"
        />
        <div
          className="stat"
        >
          <div
            className="stat-nb"
          >
            <span
              className={undefined}
              style={undefined}
            >
              10,000
            </span>
          </div>
          <div
            className="stat-nb stat-label"
          >
            <span
              className={undefined}
              style={undefined}
            >
              contributions
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="stat-container"
    >
      <div
        className="stat-box"
      >
        <div
          className="stat-icon assembl-icon-profil white"
        />
        <div
          className="stat"
        >
          <div
            className="stat-nb"
          >
            <span
              className={undefined}
              style={undefined}
            >
              10,000
            </span>
          </div>
          <div
            className="stat-nb stat-label"
          >
            <span
              className={undefined}
              style={undefined}
            >
              participants
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="stat-container"
    >
      <div
        className="stat-box"
      >
        <div
          className="stat-icon assembl-icon-participation-vote white"
        />
        <div
          className="stat"
        >
          <div
            className="stat-nb"
          >
            <span
              className={undefined}
              style={undefined}
            >
              10,000
            </span>
          </div>
          <div
            className="stat-nb stat-label"
          >
            <span
              className={undefined}
              style={undefined}
            >
              participations
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
